src/components/Navbar/index.tsx

```diff
@@ -1,78 +1,72 @@
 import Image from 'next/image';
 import Link from 'next/link';
-import React, { useEffect, useRef, useState } from 'react';
+import { useState } from 'react';
+import { FaBars, FaRegUser, FaTimes } from 'react-icons/fa';
 
 import LoginModal from '@/components/LoginModal';
-import NavbarMenuItems from '@/components/Navbar/NavbarMenuItems';
-import UserProfileButton from '@/components/Navbar/UserProfileButton';
-import UserLoginShimmer from '@/components/ShimmerEffect/UserLoginShimmer';
 import useAuthenticated from '@/hooks/useAuthenticated';
 
-const Navbar = () => {
-    const [menuOpen, setMenuOpen] = useState(false);
+import DesktopMenu from './DesktopMenu';
+import MobileMenu from './MobileMenu';
+
+const Navbar: React.FC = () => {
     const [showLoginModal, setShowLoginModal] = useState<boolean>(false);
-    const { isLoggedIn, isLoading, userData } = useAuthenticated();
+    const [isMobileMenuOpen, setIsMobileMenuOpen] = useState<boolean>(false);
 
+    const { isLoggedIn, isLoading, userData } = useAuthenticated();
     const userName = userData?.data?.userName || 'User';
     const [firstName, lastName] = userName.split(' ');
 
-    const handleMenuClick = () => {
-        setMenuOpen(!menuOpen);
+    const handleProfileClick = () => {
+        if (!isLoggedIn) {
+            setShowLoginModal(true);
+        }
     };
 
-    const renderUserProfile = () => {
-        if (isLoading) {
-            return <UserLoginShimmer />;
-        }
+    const toggleMobileMenu = () => setIsMobileMenuOpen((prev) => !prev);
 
-        return (
-            <UserProfileButton
+    const profileButton = !isLoggedIn ? (
+        <button onClick={handleProfileClick} className="text-white focus:outline-none sm:hidden z-50">
+            <FaRegUser size={24} />
+        </button>
+    ) : (
+        <button onClick={toggleMobileMenu} className="text-white focus:outline-none sm:hidden z-50">
+            {isMobileMenuOpen ? <FaTimes size={24} /> : <FaBars size={24} />}
+        </button>
+    );
```

Comment:
- please make this a component instead of storing the jsx in a variable

Reply:
fixed